server/src/main/java/org/elasticsearch/index/mapper/ParametrizedFieldMapper.java

```diff
@@ -341,5 +352,18 @@ public final void parse(String name, TypeParser.ParserContext parserContext, Map
                 iterator.remove();
             }
         }
+
+        // These parameters were previously *always* parsed by TypeParsers#parseField(), even if they
+        // made no sense; if we've got here, that means that they're not declared on a current mapper,
+        // and so we emit a deprecation warning rather than failing a previously working mapping.
+        private static final Set<String> DEPRECATED_PARAMS
+            = Set.of("store", "meta", "index", "doc_values", "boost", "index_options", "similarity");
+
+        private boolean checkForDeprecatedParameter(String propName, Version indexCreatedVersion) {
```

Comment:
should this be static?

Reply:
also should we rename to isDeprecatedParameter? I find check more ambiguous in what it does.